File: main/java/com/marsrovers/cli/SendCommandCliComponent.java
package com.marsrovers.cli;


import com.marsrovers.control.Command;
import com.marsrovers.control.MissionController;
import com.marsrovers.exceptions.RoverNotFoundException;
import com.marsrovers.io.IOHandler;

public class SendCommandCliComponent implements CliComponent{

    private final IOHandler handler;
    private final MissionController controller;

    SendCommandCliComponent(IOHandler handler, MissionController controller) {
        this.handler = handler;
        this.controller = controller;
    }


    @Override
    public void execute() {

        try {
            handler.write("Invalid commands will be ignored.");
            String roverId = handler.readWithMessage("rover_id>");
            String commandString= handler.readWithMessage("command>");

            handler.write(commandString);

            controller.send(roverId, new Command(commandString));

        } catch (RoverNotFoundException e) {
            handler.write(e.getMessage());
        }
    }
}

File: main/java/com/marsrovers/cli/SendCommandCliComponent.kt
package com.marsrovers.cli


import com.marsrovers.control.Command
import com.marsrovers.control.MissionController
import com.marsrovers.exceptions.RoverNotFoundException
import com.marsrovers.io.IOHandler

class SendCommandCliComponent internal constructor(private val handler: IOHandler, private val controller: MissionController) : CliComponent {


    override fun execute() {

        try {
            handler.write("Invalid commands will be ignored.")
            val roverId = handler.readWithMessage("rover_id>")
            val commandString = handler.readWithMessage("command>")

            handler.write(commandString)

            controller.send(roverId, Command(commandString))

        } catch (e: RoverNotFoundException) {
            handler.write(e.message!!)
        }

    }
}
